R/cube-subset.R

```diff
@@ -222,14 +223,14 @@ skipMissingCategories <- function(cube, index){
         }, visible = visible_cats, sub = index, SIMPLIFY = FALSE)
 }
 
-subsetArrayDimension <- function(dim, idx){
+subsetArrayDimension <- function(dim, idx, dim_type){
     dim$missing <- dim$missing[idx]
     dim$name <- dim$name[idx]
 
     # subset the category or item metadata
-    if (dim$references$type == "categorical") {
+    if (dim_type %in% c("categorical", "ca_categories")) {
```

Comment:
But not "mr_selections"? Aren't those categories too?

Reply:
They are, I didn't include them originally because I don't think they should ever show up here since we never let someone subset the MR selections out of the cube. But I've added them just in case that ever does happen, it doesn't hurt to have them there and I was probably overthinking it the first time.